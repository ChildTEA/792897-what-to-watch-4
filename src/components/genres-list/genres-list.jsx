import PropTypes from "prop-types";
import React from "react";
import {connect} from "react-redux";
import {ActionCreator} from "../../reducer.js";


const GenresList = ({
  filterTypes,
  activeFilter,
  onFilterItemClick,
}) => {
  return (
    <ul className="catalog__genres-list">
      {filterTypes.map((filterType) => {
        return (
          <li
            key={filterType}
            className={`catalog__genres-item ${
              activeFilter === filterType ? `catalog__genres-item--active` : ``}`}
          >
            <a
              href="#"
              className="catalog__genres-link"
              onClick={() => {
                onFilterItemClick(filterType);
              }}
            >
              {filterType}
            </a>
          </li>
        );
      })}
    </ul>
  );
};


GenresList.propTypes = {
  activeFilter: PropTypes.string.isRequired,
  filterTypes: PropTypes.arrayOf(PropTypes.string).isRequired,
  onFilterItemClick: PropTypes.func.isRequired,
};


const mapStateToProps = (state) => ({
  activeFilter: state.activeFilterType,
});

const mapDispatchToProps = (dispatch) => ({
  onFilterItemClick(filterType) {
    dispatch(ActionCreator.setFilterType(filterType));
    dispatch(ActionCreator.getMoviesByGenre());
  },

  onUserAnswer() {
    dispatch(ActionCreator.incrementStep());
  },
});


export {GenresList};
export default connect(mapStateToProps, mapDispatchToProps)(GenresList);
