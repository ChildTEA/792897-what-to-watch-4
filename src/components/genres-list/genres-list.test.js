import React from "react";
import renderer from "react-test-renderer";
import {GenresList} from "./genres-list.jsx";

const filterTypes = [`All`, `Comedy`, `Horror`, `18+`];

describe(`<GenresList />`, () => {
  it(`Should GenresList render correctcly`, () => {
    const tree = renderer
      .create(
          <GenresList
            activeFilter={filterTypes[0]}
            filterTypes={filterTypes}
            onFilterItemClick={() => {}}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
