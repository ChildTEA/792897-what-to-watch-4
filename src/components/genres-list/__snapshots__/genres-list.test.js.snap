// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenresList /> Should GenresList render correctcly 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      All
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Comedy
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Horror
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      18+
    </a>
  </li>
</ul>
`;
